treeherder/perf/tasks.py

```diff
@@ -10,3 +16,49 @@ def generate_alerts(signature_id):
     newrelic.agent.add_custom_parameter("signature_id", str(signature_id))
     signature = PerformanceSignature.objects.get(id=signature_id)
     generate_new_alerts_in_series(signature)
+
+
+@task(name='check-confirming-perf-alerts')
+def check_confirming_perf_alerts_status():
+    confirming_interval = settings.PERFHERDER_CONFIRMING_INTERVAL
+
+    def traceback_producer_job_type(perf_alert):
+        push = perf_alert.summary.push
+        signature = perf_alert.series_signature
+
+        associated_perf_datum = PerformanceDatum.objects.filter(push=push, signature=signature).first()
+        return associated_perf_datum.job.job_type if associated_perf_datum else None
+
+    def extract_pushes_to_check(many_nearby_pushes, alert_push, push_range=2):
+        many_nearby_pushes = list(many_nearby_pushes)
+
+        alert_push_idx = None
+        for index, push in enumerate(many_nearby_pushes):
+            if alert_push.id == push.id:
+                alert_push_idx = index
+                break
+
+        from_push = alert_push_idx - push_range
+        from_push = from_push if from_push > 0 else 0
+        to_push = alert_push_idx + push_range
+
+        return list(many_nearby_pushes[from_push:to_push])
+
+    confirming_alerts = PerformanceAlert.objects.filter(status=PerformanceAlert.CONFIRMING)
+    for alert in confirming_alerts:
+        alert_push = alert.summary.push
+        alert_repository = alert.series_signature.repository
+        _from, _to = alert_push.time-confirming_interval, alert_push.time+confirming_interval
+
+        job_type = traceback_producer_job_type(alert)
+        if job_type is None:
+            raise LookupError('Could not find job type for perf alert with id {}'.format(alert.id))
+
+        many_nearby_pushes = Push.objects.filter(
+            time__gte=_from, time__lte=_to, repository=alert_repository).order_by('time')
+        pushes_to_check = extract_pushes_to_check(many_nearby_pushes, alert_push)
+
+        if not Job.objects.filter(
+                push__in=pushes_to_check, job_type=job_type, autoclassify_status=Job.PENDING).exists():
```

Comment:
`autoclassify_status` means something else - and using it could give a race condition where an alert never gets confirmed. I think `state` is what you were after?

Also, what happens if a scheduled job doesn't run or gets stuck as pending forever (this happens from time to time). In that case it seems that the alert will get stuck as "confirming" and the number of alerts handled in this for loop will slowly increase over time, causing the cost of this task to increase.

Could the end goal of this feature instead be achieved in a simpler way, by instead adding a "waiting" status (which indicates the alert is not actionable yet, like "confirming" did), but then just have this task change "waiting" to "<something>" once a set amount of time has passed (eg 2 hours, which should be enough time for the jobs to have finished)?

Reply:
I like the background worker approach because it's more deterministic and you don't have to peak on every 'confirming' alert.

Reply:
The snooze feature would still be implemented as a background worker, if you thought I meant otherwise? :-)